mlflow/utils/autologging_utils/versioning.py

```diff
@@ -28,6 +28,12 @@ def _check_version_in_range(ver, min_ver, max_ver):
     return Version(min_ver) <= Version(ver) <= Version(max_ver)
 
 
+def _check_spark_version_in_range(ver, min_ver, max_ver):
+    if Version(ver) > Version(min_ver):
+        ver = _reset_minor_version(ver)
```

Comment:
The following might be less brittle (let `Version()` do the work of identifying major and minor) and is contained within this single function, since the logic isn't needed elsewhere.

```suggestion
    parsed_ver = Version(ver)
    if parsed_ver > Version(min_ver):
        ver = ".".join(parsed_ver.major, parsed_ver.minor)
```

Reply:
excellent idea. Thanks!